plugins/inputs/procstat/procstat.go

```diff
@@ -337,11 +344,67 @@ func (p *Procstat) getPIDFinder() (PIDFinder, error) {
 	return p.finder, nil
 }
 
+// Merge tags map
+func Mergetags(m, subm map[string]string) (map[string]string, error) {
+	var allm map[string]string
+	allm = m
+	for k, v := range subm {
+		_, ok := m[k]
+		if !ok {
+			allm[k] = v
+		}
+	}
+	return allm, nil
+}
+
 // Get matching PIDs and their initial tags
 func (p *Procstat) findPids() []PidsTags {
 	var pidTags []PidsTags
 
-	if p.SystemdUnit != "" {
+	if p.SupervisorUnit != "" {
+		groups, groups_tags, err := p.supervisorPIDs()
+		if err != nil {
+			pidTags = append(pidTags, PidsTags{nil, nil, err})
+			return pidTags
+		}
+		// According to the PID, find the system process number and use pgrep to filter to get the number of child processes
+		for _, group := range groups {
+			f, err := p.getPIDFinder()
+			if err != nil {
+				pidTags = append(pidTags, PidsTags{nil, nil, err})
+				return pidTags
+			}
+
+			p.Pattern = groups_tags[group].Map["pid"]
+			if p.Pattern == "" {
+				pidTags = append(pidTags, PidsTags{nil, groups_tags[group].Map, err})
+				return pidTags
+			}
+
+			pids, tags, err := p.SimpleFindPids(f)
+
+			if err != nil {
+				pidTags = append(pidTags, PidsTags{nil, nil, err})
+				return pidTags
+				// Handle situations where the PID does not exist
+			} else if len(pids) == 0 {
+				pidTags = append(pidTags, PidsTags{nil, groups_tags[group].Map, err})
+				continue
+			}
```

Comment:
As you return in the first if, you don't need the else
```suggestion
			if err != nil {
				pidTags = append(pidTags, PidsTags{nil, nil, err})
				return pidTags
			} 

			// Handle situations where the PID does not exist
			if len(pids) == 0 {
				pidTags = append(pidTags, PidsTags{nil, groups_tags[group].Map, err})
				continue
			}
```